### PR TITLE
Update to TypeScript 5.4

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Update Client to TypeScript 5.4

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/drivers/routerlicious-driver/src/shreddedSummaryDocumentStorageService.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export class ShreddedSummaryDocumentStorageService implements IDocumentStorageSe
 			},
 			async (event) => {
 				const manager = await this.getStorageManager();
-				const response = (await manager.getTree(requestVersion!.treeId)).content;
+				const response = (await manager.getTree(requestVersion.treeId)).content;
```
Comment (Josmithr) [inline packages/drivers/routerlicious-driver/src/shreddedSummaryDocumentStorageService.ts:125]: Nice!

File: packages/test/test-utils/api-report/test-utils.api.md [hunk 2]
```diff
@@ -67,9 +67,9 @@ export const createContainerRuntimeFactoryWithDefaultDataStore: (Base: typeof Co
     defaultFactory: IFluidDataStoreFactory;
     registryEntries: NamedFluidDataStoreRegistryEntries;
     dependencyContainer?: any;
-    requestHandlers?: RuntimeRequestHandler[] | undefined;
-    runtimeOptions?: IContainerRuntimeOptions | undefined;
-    provideEntryPoint?: ((runtime: IContainerRuntime) => Promise<FluidObject>) | undefined;
+    requestHandlers?: RuntimeRequestHandler[];
```
Comment (Josmithr) [inline packages/test/test-utils/api-report/test-utils.api.md:70]: Also nice!

File: .changeset/cold-jeans-invite.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+"@fluidframework/tinylicious-client": minor
+"@fluidframework/tinylicious-driver": minor
+"@fluidframework/tool-utils": minor
+"@fluid-experimental/tree": minor
+"@fluidframework/tree": minor
+"@fluid-experimental/tree-react-api": minor
+"@fluidframework/undo-redo": minor
+---
+
+Update to TypeScript 5.3
```
Comment (Josmithr) [inline .changeset/cold-jeans-invite.md:96]: Should be `5.4`

Comment (CraigMacomber) [reply .changeset/cold-jeans-invite.md]: Good catch. fixed.

**Review by Josmithr — APPROVED** [review]

**Pull request merged**